controllers/rollingupgrade_controller.go

```diff
@@ -615,6 +623,7 @@ func (r *RollingUpgradeReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	r.generatedClient = kubernetes.NewForConfigOrDie(mgr.GetConfig())
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&upgrademgrv1alpha1.RollingUpgrade{}).
+		WithOptions(controller.Options{MaxConcurrentReconciles: r.maxParallel}).
```

Comment:
This adds the option `MaxConcurrentReconciles` when starting the controller according to what is passed in the flag.